### PR TITLE
WIP: Krishna chc 188 folder structure

### PR DESCRIPTION
**Pull request by krishnakommineni**
Includes CHC-336

### CODE REVIEW

File: concordia/views.py [hunk 1]
```diff
@@ -145,6 +141,29 @@ def get_context_data(self, **kws):
         return dict(super().get_context_data(**kws), response=response)
 
 
+class ConcordiaProjectView(TemplateView):
```
Comment (acj3rd) [inline concordia/views.py]: unit test needed for this new view class

File: concordia/views.py [hunk 2]
```diff
@@ -519,3 +561,21 @@ def get(self, request, *args, **kwargs):
 
         projects = paginator.get_page(page)
         return render(self.request, self.template_name, locals())
+
+
+class FilterCollections(generics.ListAPIView):
```
Comment (acj3rd) [inline concordia/views.py]: unit test needed for this new class

File: importer/tasks.py [hunk 3]
```diff
@@ -1,92 +1,95 @@
...
 
 logger = getLogger(__name__)
 
 
-def get_request_data(url, params=None, retry_count=1):
-    response = requests.get(url, params=params)
-    if response.status_code != 200:
-        return get_request_data(url, params, retry_count + 1)
-    return response.json()
+def get_item_id_from_item_url(item_url):
```
Comment (acj3rd) [inline importer/tasks.py]: unit test needed

File: importer/tasks.py [hunk 4]
```diff
@@ -1,92 +1,95 @@
...
+    """
+    if item_url.endswith("/"):
+        item_id = item_url.split("/")[-2]
+    else:
+        item_id = item_url.split("/")[-1]
+
+    return item_id
+
+
+def get_request_data(url, params=None, timeout=120, json_resp=True, **kwargs):
```
Comment (acj3rd) [inline importer/tasks.py]: unit test needed

File: importer/tasks.py [hunk 5]
```diff
@@ -1,92 +1,95 @@
...
-        collection_url_splits = collection_url.split("?fa=")
-        collection_url = collection_url_splits[0]
-        collection_params = {"fa": collection_url_splits[1]}
-        return collection_url, collection_params
-    else:
-        return collection_url, {}
-
-
-def get_collection_item_ids(collection_name, collection_url):
+def get_collection_item_ids(collection_url, total_pages):
```
Comment (acj3rd) [inline importer/tasks.py]: changed method name needs a unit test

File: importer/tasks.py [hunk 6]
```diff
@@ -98,147 +101,132 @@ def get_collection_item_asset_urls(collection_name, item_id):
...
-            if retry_count >= 3:
-                return False
-            download_write_collection_item_assets(
-                image_url, asset_local_path, retry_count=retry_count + 1
-            )
+            logger.error("Error while writing the file to disk : %s " % str(e))
+    return False
+
+
+def get_save_item_assets(collection_name, project, item_id, item_asset_urls):
```
Comment (acj3rd) [inline importer/tasks.py]: new method needs a unit test

File: importer/tasks.py [hunk 7]
```diff
@@ -98,147 +101,132 @@ def get_collection_item_asset_urls(collection_name, item_id):
...
+
+    for idx, ciau in enumerate(item_asset_urls):
+        asset_local_path = os.path.join(item_local_path, "{0}.jpg".format(str(idx)))
+
+        download_write_collection_item_asset(ciau, asset_local_path)
 
 
 @task
-def download_write_collection_item_assets(collection_name, collection_url):
+def download_write_collection_item_assets(collection_name, project, collection_url):
```
Comment (acj3rd) [inline importer/tasks.py]: if a unit tests exists for this method, has it been rewritten top reflect the new logic? if a unit test does not exist, it needs to be written.

**Comment by elainekamlley** [comment]
@krishnakommineni do you have a aws instance I can test?

**Comment by acj3rd** [comment]
1. Unit tests for test_DeleteCollection_get and test_ExportCollectionView_get fail. The reason is the  change to urls.py hides all calls to any route with /transcribe/<value>, where <value> is     exportCSV, exportBagit, delete, report, transcription.

This will cause the existing site code to "break".

Always make sure existing units tests pass when code in the urls and views is changed.

2. Unit tests missing for
a. GET route: get_task_status/(?P<task_id>[a-zA-Z0-9-]+)
b. GET route: check_and_save_collection_assets/(?P<task_id>[a-zA-Z0-9-]+)/(?P<item_id>[a-zA-Z0-9-]+)$"
c. GET route: check_and_save_collection_assets/(?P<task_id>[a-zA-Z0-9-]+)/$",

**Pull request merged**